packages/components/src/2016-12-01/button-state.js

```diff
@@ -14,28 +14,43 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 */
+function prevState(state) {
+  return {
+    disabled: state.disabled,
+    hasFocus: state.hasFocus,
```

Comment:
Actually, we can't name a property `focus`. That's already taken. https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus

We could allow the react and angular wrappers to support an attribute called `focus`, but the state machine needs to call it something else.